charts/gateway-helm/templates/infra-manager-rbac.yaml

```diff
@@ -57,3 +57,37 @@ subjects:
 - kind: ServiceAccount
   name: 'envoy-gateway'
   namespace: '{{ .Release.Namespace }}'
+---
+apiVersion: rbac.authorization.k8s.io/v1
```

Comment:
when we chatted earlier, I failed to realize that creating a clusterRole programmatically would require us to acquire a clusterRole with a super power to create more clusterRoles 🙈 , this is not going to land well in any security audit process

Reply:
instead a common `envoy-gateway-node-reader` ClusterRole, ClusterRoleBinding and ServiceAccount is may be the way to go
are there any other alternatives ?

Reply:
I was surprised by your suggestion as well 😃.
Should we revisit doing this via EDS again?

Reply:
EDS will be tricky because it will be unique to every envoy pod, and we will need to manage an additional unique cache for this in xds server which will be tricky

another idea is to create a Secret with nodeName as key and zone as value (this can be done in the gateway-api layer and saved in the Infra IR)

so now we can either do something like
```
    env:
    - name: NODE_ZONE
      valueFrom:
        secretKeyRef:
          name: node-zone
          key: $(NODE_NAME)
```          
and if something like this isnt possible, we can mount the secret into the init container, and read it as a file

Reply:
I've got an example mutating webhook working to inject the zone to a pod at `binding` (when the node is selected) and we could do something like this:

```
    env:
    - name : ZONE
      valueFrom :
        fieldRef :
          fieldPath : metadata.labels['topology.kubernetes.io/zone']
    args:
    - "--service-zone=$(ZONE)"
```

Reply:
sure that works